### PR TITLE
Mock Package

### PR DESCRIPTION
**Pull request by f2prateek**
API compatible with existing implementation.

### CODE REVIEW

**Comment by vincepri** [comment]
LGTM

**Comment by dominicbarnes** [comment]
Is pulling in the mock library necessary at this layer? I think exposing the interface is sufficient, that way you're not locked into any specific mock library.

File: mock/mock.go [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package mock
+
+import "github.com/stretchr/testify/mock"
+
+type Client struct {
+	mock.Mock
+}
+
+func NewClient() *Client {
```
Comment (dominicbarnes) [inline mock/mock.go:13]: This `NewClient()` returns a pointer, whereas the "normal" one returns a non-pointer. Is that difference intentional?

Comment (f2prateek) [inline mock/mock.go:13]: The "normal" one returns an interface. The actual `client` type is hidden in the package (hence lowercase) - the idea being that only the interface is exposed in the public API - while giving a default implementation for consumers of the library.

**Comment by f2prateek** [comment]
> Is pulling in the mock library necessary at this layer? I think exposing the interface is sufficient, that way you're not locked into any specific mock library.

I think so - since most consumers are going to need to do some kind of mocking to test their code. By exposing just an interface in the RPC package API as we've done, consumers can choose to use their own libraries for mocking if they need to (or use no libraries and just write their own implementation). The goal of the mock package is just a standardized way to make it easier to consumers to get started with testing.

**Pull request merged**